src/mumble/AudioOutput.h

```diff
@@ -121,14 +120,16 @@ class AudioOutput : public QThread {
 	~AudioOutput() Q_DECL_OVERRIDE;
 
 	void addFrameToBuffer(ClientUser *sender, const Mumble::Protocol::AudioData &audioData);
-	void removeBuffer(const ClientUser *);
-	AudioOutputSample *playSample(const QString &filename, float volume, bool loop = false);
+	std::unique_ptr< AudioOutputToken > playSample(const QString &filename, float volume, bool loop = false);
 	void run() Q_DECL_OVERRIDE = 0;
 	virtual bool isAlive() const;
 	const float *getSpeakerPos(unsigned int &nspeakers);
 	static float calcGain(float dotproduct, float distance);
 	unsigned int getMixerFreq() const;
 	void setBufferSize(unsigned int bufferSize);
+	void setBufferPosition(AudioOutputToken *token, float x, float y, float z);
+	void removeToken(AudioOutputToken *);
```

Comment:
Why are you using a pointer to an `AudioOutputToken` here? This seems like a bug or at least like an inconsistency... We have a type wrapping a pointer to an audio buffer and here you expect a _pointer_ to this wrapper. However, the wrapper itself is pretty much designed to be handed around _by value_.

Aside from that: As you are passing the token as a _pointer_: is there a valid use-case where `nullptr` is passed as an argument (aka: is this handled in a meaningful way in the implementation)? If not, this should at the very least be a reference anyway (probably a `const` one)

Reply:
Yes. You see... C dialects are not my "native" languages. While I understand the basic concept of pointers and memory safety, these are the kind of subtleties that I would totally miss.

So the AudioOutputToken can never be null, but the underlying pointer can be. For example, if the sample creation failed. I solved this by introducing a static ``AudioOutputToken::None()`` which returns a Token object value with nullptr as AudioOutputBuffer. Makes comparision more intuitive I think, but tell me if you like this.